import Typing bug

Error 529c360e6779744861a5e5fb469b344aad295b5f:
```
ImportError while loading conftest 'D:\Old\Classes\RL_EE\VotingIsAllYouNeed\tests\conftest.py'.
tests\conftest.py:13: in <module>
    import VIAYN.samples.factory as factory
VIAYN\samples\factory.py:17: in <module>
    from VIAYN.samples.agents import (
VIAYN\samples\agents.py:31: in <module>
    class StaticBetSelectionMech(Generic[A, S], BetSelectionMechanism):
C:\Users\suhai\Anaconda3\lib\abc.py:126: in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
E   TypeError: Cannot create a consistent method resolution
E   order (MRO) for bases Generic, BetSelectionMechanism
```

to reproduce it, run pytest

--- a/VIAYN/samples/agents.py
+++ b/VIAYN/samples/agents.py
@@ -1,3 +1,8 @@
+# -*- coding: utf-8 -*-
+# @Author: Carter.Blum
+# @Date:   2020-12-06 14:44:56
+# @Last Modified by:   Suhail.Alnahari
+# @Last Modified time: 2020-12-06 14:54:12
 from abc import ABC, abstractmethod
 from copy import copy
 from typing import List, Callable, Optional, Generic
@@ -28,7 +33,7 @@ class BetSelectionMechanism(Generic[A, S], ABC):
         ...
 
 
-class StaticBetSelectionMech(Generic[A, S], BetSelectionMechanism):
+class StaticBetSelectionMech(Generic[A, S], BetSelectionMechanism [A, S]):
     def __init__(self,
             constant_bet: List[float]):
         for bet in constant_bet:

--- a/tests/test_simple_env_agents.py
+++ b/tests/test_simple_env_agents.py
@@ -2,7 +2,7 @@
 # @Author: Suhail.Alnahari
 # @Date:   2020-12-03 19:25:18
 # @Last Modified by:   Suhail.Alnahari
-# @Last Modified time: 2020-12-04 20:48:07
+# @Last Modified time: 2020-12-06 14:50:12
 
 
 from conftest import pytest, project_types,factory as fac, vote_range,np
@@ -47,7 +47,7 @@ def test_random_simple_agent_basic(random_agent_config):
     for vote,seed in random_agent_config:
         print(f"random: {vote} , {seed}")
         votingConf = fac.VotingConfigFactory.create(
-            fac.VotingConfigFactorySpec()
+            fac.VotingConfigFactorySpec(fac.VotingConfigEnum.simple)
         )
         agent = fac.AgentFactory.create(
             fac.AgentFactorySpec(
@@ -98,7 +98,7 @@ def test_random_simple_agent_forward_prediction(N):
     vote, seed = (5,0)
     print(f"random: {vote} , {seed}")
     votingConf = fac.VotingConfigFactory.create(
-        fac.VotingConfigFactorySpec()
+        fac.VotingConfigFactorySpec(fac.VotingConfigEnum.simple)
     )
     agent = fac.AgentFactory.create(
         fac.AgentFactorySpec(

